### PR TITLE
[node] v12.19.0 released

### PR DESCRIPTION
**Pull request by Semigradsky**
https://github.com/nodejs/node/releases/tag/v12.19.0
Release changes:
- crypto: add randomInt function
- module: deprecate module.parent
- deprecate process.umask() with no arguments
- async_hooks: add AsyncResource.bind utility
- assert: port common.mustCall() to assert
- fs: implement lutimes
- http: add maxTotalSockets to agent class
- http: expose host and protocol on ClientRequest
- perf_hooks: add idleTime and event loop util
- timers: allow timers to be used as primitives
- tls: make 'createSecureContext' honor more options
- zlib: add maxOutputLength option
- worker: add stack size resource limit option
- worker: add option to track unmanaged file descriptors
- worker: emit 'messagerror' events for failed deserialization
- worker: add public method for marking objects as untransferable
- worker: make FileHandle transferable

Additional changes:
- `process.umask` can take string since v12
- `options` is optional in `tls.createSecureContext` since v10
- `aborted` is boolean in `http.ClientRequest` since v11

### CODE REVIEW

**Comment by Semigradsky** [comment]
@BendingBender Hi! In this PR I have added `fs.lutimes` (it is available in node v.12 and v.14).
Tests for `graceful-fs` currently failed on these changes. Could you please take a look?

**Comment by mbelsky** [comment]
> Error: ../../../../../.dts/typescript-installs/4.1/node_modules/typescript/lib/lib.esnext.d.ts(25,21): error TS2726: Cannot find lib definition for 'esnext.weakref'.

It looks like an issue with typescript and I have same in #48581

As I see @sandersen had fixed this a few hours ago in [typescript repo](https://github.com/microsoft/TypeScript/commit/fc783a9000bbce45faa6f12e41a8691a68676e49) but I had no idea how soon the fix will be shipped to dts

**Comment by orta** [comment]
This looks good to me, for node releases I want to be a bit cautious so I've moved this PR so that one approval by anyone on this list will allow for self merging:

@Bartvds @BendingBender @microsoft @DefinitelyTyped @jkomyno @a-tarasyuk @alvis @r3nya @btoueg @BrunoScheufler @smac89 @Touffy @DeividasBakanas @eyqs @Flarna @Hannes-Magnusson-CK @KSXGitHub @hoo29 @kjin @ajafff @islishude @mwiktorczyk @mohsen1 @n-e @galkin @parambirs @eps1lon @SimonSchick @ThomasdenH @WilcoBakker @wwwy3y3 @samuela @kuehlein @j-oliveras @bhongy @chyzwar @trivikr @nguymin4 @yoursunny @qwelias @ExE-Boss @Ryan-Willpower @peterblazejewicz @addaleax @JasonHK @victorperin

**Comment by Semigradsky** [comment]
Up

File: types/node/globals.d.ts [hunk 1]
```diff
@@ -541,6 +541,7 @@ declare namespace NodeJS {
     interface Timer extends RefCounted {
         hasRef(): boolean;
         refresh(): this;
+        [Symbol.toPrimitive](): number;
```
Comment (Flarna) [inline types/node/globals.d.ts:544]: I think `clearTimeout()` should be changed to accept number because of this.

File: types/node/http.d.ts [hunk 2]
```diff
@@ -205,7 +205,9 @@ declare module "http" {
 
     // https://github.com/nodejs/node/blob/master/lib/_http_client.js#L77
     class ClientRequest extends OutgoingMessage {
-        aborted: number;
+        aborted: boolean;
+        host: string;
+        protocol: string;
```
Comment (Flarna) [inline types/node/http.d.ts:210]: I think `host` and `protocol` (maybe also others)  shoudl be readonly. Technically you can write to `host` but it is not used by http so it makes so sense.

**Review by Flarna — APPROVED** [review]

**Comment by Semigradsky** [comment]
Ready to merge

File: types/node/http.d.ts (hunk 2 above)
Comment (Semigradsky) [reply types/node/http.d.ts:210]: Looks like a lot of fields can be made readonly. I will try it later in separate PR.

**Pull request merged**